File: web/types/component.ts
import type { ComponentType } from "./enums";
import type { TagResponse } from "./tag";

export type ComponentResponse = {
  id: number;
  projectId: number;
  name: string;
  type: ComponentType;
  techStack: string | null;
  localUrl: string | null;
  repoPath: string | null;
  currentStatusNote: string | null;
  lastActivityAt: string | null;
  createdAt: string;
  updatedAt: string | null;
  isDeleted: boolean;
  deletedAt: string | null;
  tags: TagResponse[];
};

export type ComponentCreateRequest = {
  name: string;
  type: ComponentType;
  techStack?: string | null;
  localUrl?: string | null;
  repoPath?: string | null;
  currentStatusNote?: string | null;
};

export type ComponentUpdateRequest = ComponentCreateRequest;

export type ComponentStatusNoteRequest = {
  currentStatusNote: string | null;
};

File: src/DevTrack.Domain/Entities/Component.cs
using DevTrack.Domain.Enums;

namespace DevTrack.Domain.Entities;

public class Component : SoftDeletableEntity
{
    public int ProjectId { get; set; }
    public string Name { get; set; } = string.Empty;
    public ComponentType Type { get; set; } = ComponentType.Other;
    public string? TechStack { get; set; }
    public string? LocalUrl { get; set; }
    public string? RepoPath { get; set; }
    public string? CurrentStatusNote { get; set; }
    public DateTime? LastActivityAt { get; set; }

    public Project Project { get; set; } = null!;
    public ICollection<ComponentTag> ComponentTags { get; set; } = new List<ComponentTag>();
}
